src/sparseml/modifiers/obcq/base.py

```diff
@@ -61,3 +61,27 @@ class SparseGPTModifier(Modifier):
 
     def on_initialize_structure(self, state: "State", **kwargs):
         pass  # nothing needed for this modifier
+
+    def _validate_layerwise_sparisity(self):
+        if isinstance(self.sparsity, float):
+            return  # single sparsity will be applied to all layers
+
+        if not isinstance(self.compress_layers, List):
+            raise ValueError(
+                "Layer targets must be a list when specifying layer-wise"
+                f" sparsity. Got {self.compress_layers}"
+            )
+
+        if len(self.compress_layers) != len(self.sparsity):
+            raise ValueError(
+                "Number of layer targets must match the number of "
+                f"sparsities. Got {len(self.compress_layers)} layers and "
+                f"{len(self.sparsity)} sparsities"
+            )
+
+        for layer_name in self.compress_layers:
+            if "re:" in layer_name:
```

Comment:
Not sure where "re:" is rwt layer_name, if always starts with it, I'd recommend using layer_name.startswith("re:")

Reply:
Good catch, for regex its always at the beginning so startswith makes more sense!

Reply:
Good catch!